jit_decoder.go

```
@@ -0,0 +1,615 @@
+		if t != destTypes[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// cachedJITDecoder pairs a compiledRowDecoder with the exact destTypes shape
+// it was compiled for, so a cache hit can be confirmed with a cheap
+// comparison against the current Scan call's dest slice instead of
+// recomputing a string key.
+type cachedJITDecoder struct {
```

golangci-lint (2.12.2)

[error] fieldalignment: struct with 32 pointer bytes could be 16

(govet)